meeg_preprocessing/utils.py

```diff
@@ -151,15 +153,28 @@ def setup_provenance(script, results_dir, config=None, use_agg=True):
     - sets log file for sterr output
     - writes log file with runtime information
     """
+    import inspect
+    if script is None:
+        script = inspect.stack()[1][1]
+        if script[0] == '<':
+            script = '__main__'
```

Comment:
> Why don't you put a debugger at the part of your script that's under development and run the script up to that point.

Mmmh, that would increase the number of lines; the aim is to make life easier.

I agree that there seems to be something dangerous with the inspect approach, but I can't quite pinpoint what it is either.

I think we'll remain in disagreement on the style; I strongly believe that we need to be able to do some dirty copy and paste in the console when we create a new script. running scripts with run etc is annoying because python (unlike matlab) keeps the modules in memory and does not systematically take into account the changes in the file that occur after loading. (You can make it so, I know, but this require extra command lines)